internal/serviceapi/projectsettingsapi/resource_test.go

```diff
@@ -0,0 +1,108 @@
+package projectsettingsapi_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/acc"
+)
+
+const resourceName = "mongodbatlas_project_settings_api.test"
+
+func TestAccProjectSettingsAPI_basic(t *testing.T) {
+	var (
+		projectID = acc.ProjectIDExecution(t)
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acc.PreCheckBasic(t) },
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		Steps: []resource.TestStep{
+			{
+				Config: config(projectID, false),
+				Check:  check(projectID, false),
+			},
+			{
+				Config: config(projectID, true),
+				Check:  check(projectID, true),
+			},
+			{
+				Config: config(projectID, false),
+				Check:  check(projectID, false),
+			},
```

Comment:
nit: Could be could to add a step where some or all attributes are not defined, given they are O+C values in state would not change.